File: Programming/3 семестр/JAVA/sem3_kr2/src/Tree.java
public abstract class Tree implements Comparable<Tree> {
    private String name;
    private Integer numberOfLeaves;
    private Type type;

    public enum Type {
        DECIDUOUS, CONIFER;
        private Type() {
        }
    }

    public Tree() {
        name = "";
        numberOfLeaves = 0;
    }

    public Tree(String name, int  numberOfLeaves, Type type) throws MyException {
        if (name == null ||  numberOfLeaves <0 || type == null)
            throw new MyException("THE DATA IS INCORRECT");
        this.name= name;
        this.numberOfLeaves = numberOfLeaves;
        this.type = type;
    }

    public int getNumberOfLeaves() {
        return numberOfLeaves;
    }

    public String getName() {
        return name;
    }

    public Type getType() {
        return type;
    }

    public void setNumberOfLeaves(int numberOfLeaves) {
        this.numberOfLeaves = numberOfLeaves;
    }

    public void setType(Type type) {
        this.type = type;
    }

    public void setName(String name) {
        this.name = name;
    }

    public abstract void print();
    @Override
    public String toString() {
        return new String("[" + this.getClass().getName() + "] "
                + this.getName() + ": \n*"
                + this.getNumberOfLeaves()+ " leaves"+ "\n*"
                + this.getType());
    }

    @Override
    public int compareTo(Tree tree) {
        if (this.getNumberOfLeaves()==tree.getNumberOfLeaves())
            return tree.getName().compareTo(this.getName());//по убыванию
        return tree.getNumberOfLeaves()-this.getNumberOfLeaves();
    }

    @Override
    public boolean equals(Object obj) {
        Tree tree = (Tree) obj;
        return this.name.equals(tree.name) &&
                this.numberOfLeaves==tree.numberOfLeaves &&
                this.type.equals(tree.type);
    }
}

File: Programming/2 семестр/STL/LAb3/LAb3/Tree.cpp
#include "Tree.h"

Tree::Tree(string name, int age, Type1 type)
{
	this->name = name;
	this->age = age;
	this->type = type;
}

Type1 Tree::getType()
{
	return type;
}


string Tree::getName()
{
	return name;
}

int Tree::getAge()
{
	return age;
}

void Tree::setName(string name)
{
	this->name = name;
}

void Tree::setAge(int age)
{
	this->age = age;
}

void Tree::setType(Type1 type)
{
	this->type = type;
}

Type2 Tree::getType2()
{
	return Type2 ::L;
}
